docs/general-concepts/logging.md

```diff
@@ -0,0 +1,231 @@
+---
+title: Logging
+---
+
+Logging
+=======
+
+This section describes the Joomla logging framework, which enables extensions to log messages to files, etc.
+
+You can use this feature for debugging your extensions, 
+but more importantly for troubleshooting problems with your extensions in live operation.
+For example, you can logon to a site where your extension is live, navigate through the site,
+and view your extension's logged messages via a debug console, while other users of the site are unaffected.
+
+There isn't a sample extension to download to demonstrate the functionality,
+but you can easily use one of the [module tutorial](../building-extensions/modules/module-development-tutorial/index.md) steps
+as the basis for experimenting with the functions described below.
+
+Note that Joomla logging is different from Joomla Action Logs (which isn't covered here).
+
+## Basic Logging
+
+To log a message you use the static `add()` function of the Joomla\CMS\Log\Log class. For example: 
+
+```php
+use Joomla\CMS\Log\Log;
+...
+Log::add('my error message', Log::ERROR, 'my-extension-category');
+```
+
+The parameters are
+
+- A message string - whatever you like. 
+You can use language translation with these, for example, `Joomla\CMS\Language\Text::_('MY_EXTENSION_ERR_MSG')` 
+(provided you have loaded your extension's language constants). 
+You can also display the values of variables, provided you convert them into a string format. 
+
+- A priority, which can be one of Log::EMERGENCY, Log::ALERT, Log::CRITICAL, Log::ERROR, Log::WARNING, Log::NOTICE, Log::INFO, Log::DEBUG.
+These are based on the standard syslog / RFC 5424 severity levels – see the [wikipedia article on syslog](https://en.wikipedia.org/wiki/Syslog),
+and defined in libraries/src/Log/Log.php.
+
+- A category, which is just a text string. 
+You can define whatever categories you like, but it is best to define them to avoid possible clashes with those of other extensions.
+Using your extension's name is fine.
+
+:::note
+If you log a message with 'jerror' as the category then the message will also be displayed in the message area on the web page.
+:::
+
+The logging is controller by the administrator functionality, so to see the logged message you must switch logging on.
+
+In the admin Global Configuration / Logging tab:
+
+- ensure that the path to log folder is set (by default it's an absolute pathname pointing to administrator/logs). 
+
+- set the Log Almost Everything switch to Yes. 
```

Comment:
This part is often misunderstood.
Logging configuration allows 2 type of logging and they independent:
1. "**Log Almost Everything**" - logs everything except logs with "deprecation" category. And store result in `everything.php` (as you already wrote).
2. "**Custom logging**" - this will log only specified categories in to `custom-logging.php`. To enable it just need to add categories you want to log.

So to enable logging User NOT required to enable "Log Almost Everything", 
they can choose use **Log Almost Everything** or use **Custom logging**. These two is independent.